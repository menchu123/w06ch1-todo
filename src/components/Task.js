const Task = () => {
  return (
    <div className="task form-check">
      <input
        type="checkbox"
        name="isDone"
        id="isDone"
        className="form-check-input"
      />
      <label htmlFor="isDone" className="form-check-label">
        Lavar los platos
      </label>
    </div>
  );
};

export default Task;
