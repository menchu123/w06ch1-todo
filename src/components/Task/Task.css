.task__left {
  width: 100%;
}

.task__text-input {
  width: 85%;
}

.form {
  width: 100%;
}
