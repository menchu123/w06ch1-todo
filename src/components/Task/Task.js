import { useCallback, useEffect, useState } from "react";
import useTasks from "../../hooks/useTasks";
import "./Task.css";

const Task = ({ task }) => {
  const { deleteTask, updateTask } = useTasks();

  const onDelete = (event) => {
    event.preventDefault();
    deleteTask(task.id);
  };

  const initialCheck = task.done;
  //console.log(initialCheck);
  const [isChecked, setIsChecked] = useState(initialCheck);
  //console.log("esto otro" + isChecked);
  const initialData = task.task;

  const [taskData, setTaskData] = useState(initialData);
  const [isEditing, setisEditing] = useState(false);

  const onChange = (event) => {
    event.preventDefault();
    setTaskData(event.target.value);
  };

  useEffect(() => {});
  const changeChecked = (event) => {
    setIsChecked(!isChecked);
    console.log("esto" + isChecked);
    // const newTask = {
    //   ...task,
    //   done: isChecked,
    // };
    // updateTask(newTask);
  };

  useEffect(() => {
    const newTask = {
      ...task,
      done: isChecked,
    };
    updateTask(newTask);
  }, [updateTask, isChecked]);

  const onUpdate = (event) => {
    event.preventDefault();
    const newTask = {
      id: task.id,
      task: taskData,
      done: task.done,
    };

    if (taskData === "") {
      deleteTask(task.id);
    } else {
      updateTask(newTask);
      setisEditing(false);
      if (event.target.querySelector(".task__text")) {
        event.target.querySelector(".task__text").blur();
      }
    }
  };

  return (
    <li className="task form-check list-group-item d-flex justify-content-between">
      <div className="task__left">
        <input
          type="checkbox"
          checked={isChecked}
          onChange={changeChecked}
          name="task-check"
          id="task-check"
          className="form-check-input"
        />
        <form
          className="form d-flex justify-content-between"
          onSubmit={isEditing ? (event) => onUpdate(event) : onDelete}
        >
          <div className="form-group task__text-input">
            <input
              type="text"
              className="task__text form-control border-0"
              value={taskData}
              onChange={(event) => onChange(event)}
              onFocus={() => setisEditing(true)}
              onBlur={(event) => onUpdate(event)}
            ></input>
          </div>
          <button
            className={`task__delete btn btn-${
              isEditing ? "primary" : "danger"
            } pull-right`}
          >
            {isEditing ? "Edit" : "-"}
          </button>
        </form>
      </div>
    </li>
  );
};

export default Task;
