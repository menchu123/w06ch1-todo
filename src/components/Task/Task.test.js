import { render, screen } from "@testing-library/react";
import Task from "./Task";
import ReactTestRenderer from "react-test-renderer";

describe("Given a Task component", () => {
  describe("When it is passed a 'cool' text", () => {
    test("Then it should render an input with a 'cool' value", () => {
      const value = "cool";

      const task = {
        id: 3,
        task: value,
        done: true,
      };

      render(<Task task={task} />);

      const taskComponent = screen.getByRole("textbox");

      expect(taskComponent).toHaveValue(value);
    });
  });

  describe("When it recives and object", () => {
    test("then it should render a task with the task text inside", () => {
      const task = {
        id: 3,
        task: "Sacar a Elsa",
        done: true,
      };

      const taskComponent = ReactTestRenderer.create(<Task task={task} />);
      expect(taskComponent.toJSON()).toMatchSnapshot();
    });
  });
});
