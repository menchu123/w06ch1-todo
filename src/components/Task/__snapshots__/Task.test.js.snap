// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Task component When it recives and object then it should render a task with the task text inside 1`] = `
<li
  className="task form-check list-group-item d-flex justify-content-between"
>
  <div
    className="task__left"
  >
    <input
      checked={true}
      className="form-check-input"
      id="task-check"
      name="task-check"
      onChange={[Function]}
      type="checkbox"
    />
    <form
      className="form d-flex justify-content-between"
      onSubmit={[Function]}
    >
      <div
        className="form-group task__text-input"
      >
        <input
          className="task__text form-control border-0"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="text"
          value="Sacar a Elsa"
        />
      </div>
      <button
        className="task__delete btn btn-danger pull-right"
      >
        -
      </button>
    </form>
  </div>
</li>
`;
